### PR TITLE
[AscendNPU-IR] feat: add brc mechanism for T.Parallel

### PR DESCRIPTION
**Pull request by endaiHW**
1. support brc mechanism in nested parallel
2. support bufferLoad scenario in if_then_else(mask cond)
3. fix bug when sub's lhs is scalarLike bufferLoad

### CODE REVIEW

File: src/transform/npu_loop_vectorize.cc [hunk 1]
```diff
@@ -46,738 +46,781 @@ using namespace tir;
 
 class LoopDecompose : public StmtMutator {
 private:
-  // Temporary buffers of subexpression results when processing compound expressions
+  // 当前 block 内申请的所有临时 buffer，在 VisitStmt_(BlockNode) 中统一注册
   std::vector<Buffer> tmp_buffers;
-  // Recursion depth when processing compound expressions recursively
+
+  // 记录当前嵌套表达式的递归深度，通过 DepthGuard 管理
```
Comment (CeleNewYear) [inline src/transform/npu_loop_vectorize.cc]: Please use English comments.

File: format.sh [hunk 2]
```diff
@@ -5,15 +5,19 @@
```
Comment (CeleNewYear) [inline format.sh]: Why don't we follow the community's norms?

File: testing/npuir/parallel_feat/test_if_then_else_cmp_cond.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import os
+import argparse
+import torch
+
```
Comment (CeleNewYear) [inline testing/npuir/parallel_feat/test_if_then_else_cmp_cond.py]: Please use pytest mode test case

File: testing/npuir/parallel_feat/test_if_then_else_buffer_cond.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
+# Copyright (c) Huawei Technologies Co., Ltd. 2025.
+#
+# This unit test verifies the support of T.Parallel for the following scenarios:
+# 1. Vectorization of mask generation
+# 2. Correct vectorization of 2D T.Parallel with if_then_else
+# 3. Correct handling of broadcast-style access: mask[b_i] used in (h_i, b_i)
```
Comment (CeleNewYear) [inline testing/npuir/parallel_feat/test_if_then_else_buffer_cond.py]: Please use pytest mode testcase

File: testing/npuir/parallel_feat/test_brc_in_nested_parallel.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Huawei Technologies Co., Ltd. 2025.
```
Comment (CeleNewYear) [inline testing/npuir/parallel_feat/test_brc_in_nested_parallel.py]: Please use pytest mode

File: src/transform/npu_loop_vectorize.cc [hunk 6]
```diff
@@ -1086,21 +1183,21 @@ class LoopVectorize : public StmtMutator {
...
+  Stmt VisitStmt_(const ForNode *op) final {
     // try vectorize only when marked as parallel
     if (op->kind != ForKind::kParallel) {
       return StmtMutator::VisitStmt_(op);
     }
 
     // recursive processing
     Stmt body = op->body;
-    if (const auto* forNode = body.as<ForNode>()) {
+    if (const auto *forNode = body.as<ForNode>()) {
```
Comment (CeleNewYear) [inline src/transform/npu_loop_vectorize.cc]: Is it modified to comply with the clean code specifications of the community?

**Review by CeleNewYear — APPROVED** [review]

File: src/transform/npu_loop_vectorize.cc [hunk 7]
```diff
@@ -46,738 +46,805 @@ using namespace tir;
...
-    if (IsCmpOps(op_name)) {
-      std::string str = op_name;
-      std::transform(str.cbegin(), str.cend(), str.begin(), to_lowercase);
-      StringImm cmp_mode(str);
-      args.push_back(cmp_mode);
-    }
-    PrimExpr call = Call(DataType::Void(), op_type, args);
-    return Evaluate(call);
+  // Create a named buffer with the specified shape / scope / dtype.
+  Buffer CreateTempBufferWithShape(const Buffer &ref,
```
Comment (dazuozcy) [inline src/transform/npu_loop_vectorize.cc]: ref参数没有使用到，可以删除

**Review by dazuozcy — CHANGES_REQUESTED** [review]

File: src/transform/npu_loop_vectorize.cc [hunk 8]
```diff
@@ -46,738 +46,805 @@ using namespace tir;
...
+                                   const Array<PrimExpr> &shape, DataType dtype,
+                                   const std::string &name,
+                                   const std::string &scope) {
+    Var buf_var(name, PointerType(PrimType(dtype), scope));
+    return Buffer(buf_var, dtype, shape, {}, PrimExpr(0), name, 0, 0, kDefault);
```
Comment (dazuozcy) [inline src/transform/npu_loop_vectorize.cc]: The `dtype` of the output buffer is not necessarily the same as the input's. For example, in `cmp-like` ops, `dtype` of the output buffer is fixed to be `bool`. So, the `dtype` here cannot directly use the input's.
Refer to the old implementation of `CreateTempBuffer` to supplement the considerations for this scenario.

**Review by dazuozcy — DISMISSED** [review]
